test/runtests.jl

```diff
@@ -1,11 +1,104 @@
 using Proj4
-using Base.Test
+using FactCheck
 
 # Some very basic sanity checking
 wgs84 = Projection("+proj=longlat +datum=WGS84 +no_defs")
 utm56 = Projection("+proj=utm +zone=56 +south +datum=WGS84 +units=m +no_defs")
 
 # Reference data computed using GeographicLib's GeoConvert tool
-@test maximum(transform(wgs84, utm56, [150 -27 0]) .-
-              [202273.912995055 7010024.033113679 0]) < 1e-6
+@fact transform(wgs84, utm56, [150 -27 0]) -->
+    roughly([202273.912995055 7010024.033113679 0], 1e-6)
 
+dup_wgs84 = Projection("+proj=longlat +ellps=WGS84 +datum=WGS84 +no_defs")
+@fact transform(wgs84, dup_wgs84, [150 -27 0]) -->
+    roughly([150 -27 0], 1e-6)
+@fact transform(dup_wgs84, wgs84, [150 -27 0]) -->
+    roughly([150 -27 0], 1e-6)
+
+# Taken from https://github.com/proj4js/proj4js/blob/master/test/test.js
+sweref99tm = Projection("+proj=utm +zone=33 +ellps=GRS80 +towgs84=0,0,0,0,0,0,0 +units=m +no_defs")
+rt90 = Projection("+lon_0=15.808277777799999 +lat_0=0.0 +k=1.0 +x_0=1500000.0 +y_0=0.0 +proj=tmerc +ellps=bessel +units=m +towgs84=414.1,41.3,603.1,-0.855,2.141,-7.023,0 +no_defs")
+@fact transform(sweref99tm, rt90, [319180, 6399862]) -->
+    roughly([1271137.927154 6404230.291456], 1e-6)
+@fact transform(sweref99tm, rt90, [319480 6397862;
+                                   329200 6599800]) -->
+    roughly([1271414.272854 6402225.564811;
+             1283568.895883 6604160.216834], 1e-6)
+@fact transform(sweref99tm, rt90, [319480 6397862;
+                                   329200 6599800;
+                                   319480 6397862;
+                                   329200 6599800]) -->
+    roughly([1271414.272854 6402225.564811;
+             1283568.895883 6604160.216834;
+             1271414.272854 6402225.564811;
+             1283568.895883 6604160.216834], 1e-6)
+@fact transform(rt90, sweref99tm, [1271414.272854 6402225.564811;
+                                   1283568.895883 6604160.216834;
+                                   1271414.272854 6402225.564811;
+                                   1283568.895883 6604160.216834]) -->
+    roughly([319480 6397862;
+             329200 6599800;
+             319480 6397862;
+             329200 6599800], 1e-6)
+
+# lla_point = [1.290270 103.851959 0.0]
+
+for proj_string in values(Proj4.epsg)
+   proj = Projection(proj_string)
+   proj_string1 = Proj4._get_def(proj)
+   proj1 = Projection(proj_string1)
+   proj_string2 = Proj4._get_def(proj1)
+   proj2 = Projection(proj_string2)
+   proj_string3 = Proj4._get_def(proj2)
```

Comment:
It's common that people will want to get the projection string from a Projection. I'll push a `proj4string` method (name taken from python's basemap: https://github.com/matplotlib/basemap/search?q=proj4string). See if you like it.